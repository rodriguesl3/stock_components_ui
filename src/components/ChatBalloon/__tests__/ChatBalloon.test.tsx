import React from 'react';
import { shallow } from 'enzyme';
import ChatBalllon from '../ChatBalloon';
describe('ChatBalloon', () => {
  it('it should render sender correctly', () => {
    const expectText = 'this is a test';
    const timeExpect = '08:00 am';
    const submitType = 'sender';
    const wrapper = shallow(
      <ChatBalllon content={expectText} time={timeExpect} type={submitType} />,
    );
    const content = wrapper.find(`[type="${submitType}"]`).text();

    const timeContent = wrapper.find('[data-info="timer"]').text();
    expect(wrapper).toMatchSnapshot();

    expect(content).toEqual(expectText);
    expect(timeContent).toEqual(timeExpect);
  });

  it('it should render listener correctly', () => {
    const expectText = 'this is a test';
    const timeExpect = '08:00 am';
    const submitType = 'listener';
    const wrapper = shallow(
      <ChatBalllon content={expectText} time={timeExpect} type={submitType} />,
    );
    const content = wrapper.find(`[type="${submitType}"]`).text();

    const timeContent = wrapper.find('[data-info="timer"]').text();
    expect(wrapper).toMatchSnapshot();

    expect(content).toEqual(expectText);
    expect(timeContent).toEqual(timeExpect);
  });
});
