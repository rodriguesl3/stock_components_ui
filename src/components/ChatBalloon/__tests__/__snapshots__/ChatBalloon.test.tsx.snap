// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBalloon it should render listener correctly 1`] = `
<div>
  <styled.div
    type="listener"
  >
    this is a test
  </styled.div>
  <styled.span
    data-info="timer"
  >
    08:00 am
  </styled.span>
</div>
`;

exports[`ChatBalloon it should render sender correctly 1`] = `
<div>
  <styled.div
    type="sender"
  >
    this is a test
  </styled.div>
  <styled.span
    data-info="timer"
  >
    08:00 am
  </styled.span>
</div>
`;
